### PR TITLE
Rename cursor/increment

### PR DESCRIPTION
**Pull request by Alldred**
Renamed cursor to bucket
Renamed increment to hit
add_goals() now has boolean flags for illegal/ignore. No need for users to know magic numbers
Update documentation

### CODE REVIEW

File: bucket/bucket.py [hunk 1]
```diff
@@ -1,9 +1,10 @@
 # SPDX-License-Identifier: MIT
 # Copyright (c) 2023 Vypercore. All Rights Reserved
 
-class Cursor:
-    # This class takes all the axes and their values
-    # and inrements the coverage for that cross
+class Bucket:
+    '''
+    This class is used for incrementing the hit count on a given bucket
+    '''
```
Comment (Kotarski) [inline bucket/bucket.py:7]: Totally optional, but I find it really useful to have an example use up here for reference in the code. I think the syntax is (note double colon):
```
Blah blah blah, for example::

    code
```

File: bucket/coverpoint.py [hunk 2]
```diff
@@ -75,28 +75,40 @@ def all_axis_value_combinations(self):
...
+        assert sum(illegal, ignore, (target is not None)) <= 1,
+            f"Only one option may be chosen: illegal, ignore or target"
+        
+        if illegal:
+            target = -1
+        elif ignore:
+            target = 0
+        elif target is not None:
+            # This shouldn't be hardcoded, something that can be overridden would be good
+            target = 10
```
Comment (Kotarski) [inline bucket/coverpoint.py:100]: Looks like a mistake here, surely:
```
elif target is None:
    target = 10
else:
    assert target > 0, "...."
```

**Review by Kotarski — CHANGES_REQUESTED** [review]

**Review by Kotarski — APPROVED** [review]

File: bucket/bucket.py [hunk 3]
```diff
@@ -17,51 +20,56 @@ def __call__(self):
...
-        if cursor_goal.target > 0:
-            self.parent.increment_hit_count(cursor_tuple)
-        elif cursor_goal.target < 0:
-            print(f"Illegal bucket '{self.parent.name}.{cursor_goal.name}' hit!")
-            print(f"  Cursor: {dict(zip(self.parent.axis_names, list(cursor_tuple), strict=True))}")
+        # Else the bucket hit count is incremented
+        if bucket_goal.target > 0:
+            self.parent._increment_hit_count(axis_value_tuple)
+        elif bucket_goal.target < 0:
+            print(f"Illegal bucket '{self.parent.name}.{bucket_goal.name}' hit!")
```
Comment (Intuity) [inline bucket/bucket.py:68]: Could we move to use `logging` in the long run?

**Review by Intuity — APPROVED** [review]

File: bucket/bucket.py (hunk 3 above)
Comment (Alldred) [reply bucket/bucket.py:68]: I'll add it to the roadmap

Comment (Alldred) [reply bucket/bucket.py:68]: https://github.com/VyperCore/bucket/pull/23

Comment (Alldred) [reply bucket/bucket.py:68]: https://github.com/VyperCore/bucket/pull/23

**Pull request merged**